core/io/ip.cpp

```diff
@@ -179,7 +183,8 @@ IP_Address IP::get_resolve_item_address(ResolverID p_id) const {
 
 	if (resolver->queue[p_id].status != IP::RESOLVER_STATUS_DONE) {
 		ERR_EXPLAIN("Resolve of '" + resolver->queue[p_id].hostname + "'' didn't complete yet.");
-		ERR_FAIL_COND_V(resolver->queue[p_id].status != IP::RESOLVER_STATUS_DONE, IP_Address());
+		resolver->mutex->unlock();
+		return IP_Address();
```

Comment:
If I understood how error functions work, either use `ERR_FAIL_V(IP_Address());` or change `ERR_EXPLAIN(...)` to `ERR_PRINT()`, otherwise the error won't be printed (and might interfere with future errors).

I would use the first solution (`ERR_EXPLAIN`, `ERR_FAIL_V`) above too (where you are using `ERR_PRINT`), since it make it clearer that the return is an error, but I don't think we have a convention for this ( @akien-mga do we? ).
So it's up to you which solution to choose.

Beside this small thing, the PR looks now good to me.